src/main/java/carpet/patches/TickSyncedBorderExtent.java

```diff
@@ -0,0 +1,160 @@
+package carpet.patches;
+
+import carpet.CarpetServer;
+import carpet.helpers.TickSpeed;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.Mth;
+import net.minecraft.world.level.border.BorderChangeListener;
+import net.minecraft.world.level.border.BorderStatus;
+import net.minecraft.world.level.border.WorldBorder;
+import net.minecraft.world.phys.shapes.BooleanOp;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+import java.util.Arrays;
+import java.util.OptionalDouble;
+
+/**
+ * This class is essentially a copy of {@link net.minecraft.world.level.border.WorldBorder.MovingBorderExtent}
+ * but instead of using real time to lerp the border
+ * this class uses the in game ticks.
```

Comment:
Wouldn't it be better to extend/mixin to it then?

Reply:
Mixing into the class I would essentially be overwriting all the methods - slashing compatability. My original implementation did actually do this, but I decided against this.
I also tried extended the MovingBorder class, but I don't think that is a good idea; there will be unused fields which will not be accurate for the behaviour of the class: `lerpDuration` and `lerpDuration`, this would also require another AW.